docs/SIL.rst

```diff
@@ -2851,9 +2852,12 @@ Initializes or reassigns a weak reference.  The operand may be ``nil``.
 
 If ``[initialization]`` is given, the weak reference must currently either be
 uninitialized or destroyed.  If it is not given, the weak reference must
-currently be initialized and after the evaluation the value that was originally
-referenced by the weak reference will have its weak reference count decremented
-by 1.
+currently be initialized. After the evaluation:
+
+* The value that was originally referenced by the weak reference will have
+  its weak reference count decremented by 1.
+* If the optional is non-nil, the reference in the optional's strong reference count is not
+  affected.
```

Comment:
The "if" clause is unfortunate here; it ends up suggesting that the reference count is affected if the optional is nil, which is of course not what you mean.  Just "The new reference's weak reference is incremented, and its strong reference count is not affected." should be clear enough.  If you want to be explicit about what happens in the `nil` case, you should do that for all the places you're modifying.